### DISCUSSION TITLE
Split up app URL with middleware.

### DISCUSSION THREAD

**Discussion by shahensaifullah** [Neoteroi/BlackSheep#312] [Q&A] [answered]
there are two end user, merchant and public. Merhcant have different auth condition and user have diffrent auth condition with different route prefix. For merchant, there are many section, product, shop and so on. I want to split the merchant into different section. How to split the router with different condition.

**Comment by RobertoPrevato** [comment, marked as answer]
Hi @shahriar350 
Sorry for replying so late. `blacksheep` has built-in support for the scenario you describe: I would define different authorization policie, one for regular users and one for merchants.

The feature is described here: https://www.neoteroi.dev/blacksheep/authorization/

In practice you could use the same way to _authenticate_ users (e.g. using cookie or JWT Bearer authentication after sign-in) and then you would configure two _authorization policies_ applying different kind of requirements. There is an example here https://www.neoteroi.dev/blacksheep/authorization/#defining-an-authorization-policy-that-checks-users-claims
